plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/NugetGalleryFacetImpl.java

```diff
@@ -340,6 +343,13 @@ public void put(final InputStream inputStream) throws IOException, NugetPackageE
         getEventBus().post(new ComponentUpdatedEvent(component, getRepository()));
       }
     }
+
+    if (componentId != null) {
+      try (StorageTx storageTx = openStorageTx()) {
+        maintainAggregateInfo(storageTx, componentId);
+        storageTx.commit();
```

Comment:
doesn't this put the component and metadata aggregate updates in separate commits now? I would think they should both be in the same commit, if possible, or at least ensure that they're both committed in the same transaction